compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIETileAndFuse.cpp

```diff
@@ -31,163 +30,75 @@ namespace mlir::iree_compiler::AMDAIE {
 
 namespace {
 
-/// Starting from `op` walk all operands backwards to find all
-/// potentially fusable operations, i.e. operations that implement
-/// the `TilingInterface`.
-static void collectTiledAndFusedOps(Operation *rootOp,
-                                    llvm::SmallDenseSet<Operation *> &result) {
-  SmallVector<Operation *> worklist;
-  worklist.push_back(rootOp);
-  result.insert(rootOp);
-  while (!worklist.empty()) {
-    Operation *current = worklist.pop_back_val();
-    for (OpOperand &operand : current->getOpOperands()) {
-      Operation *producer = operand.get().getDefiningOp();
-      if (!producer || !isa<TilingInterface>(producer) ||
-          result.count(producer))
-        continue;
-      worklist.push_back(producer);
-      result.insert(producer);
-    }
-  }
-}
-
-LogicalResult applyTileAndFuseUsingSCFFor(RewriterBase &rewriter,
-                                          Operation *rootOp,
-                                          DominanceInfo &dominanceInfo,
-                                          scf::SCFTilingOptions options) {
-  llvm::SmallDenseSet<Operation *> origTiledAndFusedOps;
-  collectTiledAndFusedOps(rootOp, origTiledAndFusedOps);
-  auto isIgnoredUser = [&](Operation *user) {
-    return origTiledAndFusedOps.count(user) || isa<tensor::DimOp>(user);
-  };
-  FailureOr<scf::SCFTilingResult> tilingResult =
-      tileUsingSCFForOp(rewriter, cast<TilingInterface>(rootOp), options);
-  if (failed(tilingResult)) {
-    return failure();
+LogicalResult applyTileAndFuse(RewriterBase &rewriter, TilingInterface rootOp,
+                               DominanceInfo &dominanceInfo,
+                               scf::SCFTilingOptions options, bool useSCFFor,
+                               bool useFusion) {
+  if (!useSCFFor) {
```

Comment:
This can just be set in the caller, and you can just pass the options object by `const &`